packages/pg-connection-string/index.js

```diff
@@ -87,15 +87,26 @@ function parse(str) {
       break
     }
     case 'prefer':
-    case 'require':
-    case 'verify-ca':
-    case 'verify-full': {
-      break
-    }
     case 'no-verify': {
       config.ssl.rejectUnauthorized = false
       break
     }
+    case 'require': {
+      if (config.sslrootcert) {
+        // If a root CA is specified, behavior of `sslmode=require` will be the same as that of `verify-ca`
+        config.ssl.checkServerIdentity = function () {}
+      } else {
+        config.ssl.rejectUnauthorized = false
+      }
+      break
+    }
+    case 'verify-ca': {
+      config.ssl.checkServerIdentity = function () {}
```

Comment:
On second thought, even though libpq has and has documented this same problem, maybe we should enforce that `config.ssl.ca` is truthy here.

```suggestion
      if (!config.ssl.ca) {
        throw new Error('sslmode=verify-ca requires specifying a CA with sslrootcert')
      }
      config.ssl.checkServerIdentity = function () {}
```

The main issue with this would be for anyone who wants to add a `ca` key to the returned `config.ssl` *after* parsing. I don’t think a warning is the way to go (too easy for those who need it to miss or ignore, and annoying for those who don’t need it to suppress). Would adding a whole options parameter to support this (`parse(str, {deferSslCa: true})` – probably a better name option) be too much? (Related issue, merging configuration with a connection string is hard in general: #3327)